omod/src/main/webapp/resources/scripts/field/fingerprintM2sys.js

```diff
@@ -1,73 +1,81 @@
-    jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/getStatus.action');
-    //TODO add success method
+	jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/getStatus.action');
+	//TODO add success method
 }
 
-function m2sysUpdate(idValue) {
-    jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/update.action',
-        {
-            id: idValue
-        }
-    )
-    .success(function(data) {
-        if (data['success'] == true) {
-            $("#fingerprintStatus").text("Success!");
-            $("#fingerprintError").text("");
-        } else {
-            $("#fingerprintStatus").text("Failed!");
-            $("#fingerprintError").text(data['message']);
-        }
-     });
+function m2sysUpdate(idValue, button) {
+	toggleFingerprintButtonDisplay(button);
+	jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/update.action',
+		{
+			id: idValue
+		}
+	)
+	.always(function() {
+		toggleFingerprintButtonDisplay(button);
+	})
+	.success(function(data) {
+		if (data['success'] == true) {
+			$("#fingerprintStatus").text("Success!");
+			$("#fingerprintError").text("");
+		} else {
+			$("#fingerprintStatus").text("Failed!");
+			$("#fingerprintError").text(data['message']);
+		}
+	 });
 }
 
 function m2sysUpdateSubjectId(oldIdValue, newIdValue) {
-    jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/updateSubjectId.action',
-        {
-            oldId: oldIdValue,
-            newId: newIdValue
-        }
-    );
-    //TODO add success method
+	jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/updateSubjectId.action',
+		{
+			oldId: oldIdValue,
+			newId: newIdValue
+		}
+	);
+	//TODO add success method
 }
 
 function m2sysSearch(idValue) {
-    jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/search.action',
-        {
-            id: idValue
-        }
-    );
-    //TODO add success method
+	jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/search.action',
+		{
+			id: idValue
+		}
+	);
+	//TODO add success method
 }
 
 function m2sysLookup(idValue) {
-    jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/lookup.action',
-        {
-            id: idValue
-        }
-    );
-    //TODO add success method
+	jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/lookup.action',
+		{
+			id: idValue
+		}
+	);
+	//TODO add success method
 }
 
 function m2sysDelete(idValue) {
-    jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/delete.action',
-        {
-            id: idValue
-        }
-    );
-    //TODO add success method
+	jq.getJSON('/' + OPENMRS_CONTEXT_PATH + '/registrationapp/field/fingerprintM2sys/delete.action',
+		{
+			id: idValue
+		}
+	);
+	//TODO add success method
 }
```

Comment:
shouldn't a new line sign be added at the end?

https://stackoverflow.com/questions/4700312/empty-new-line-at-the-end-of-the-java-source-files/14637072#14637072

Reply:
Yes, I can do it